src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -610,11 +610,6 @@ protected override Stream OnInitiateUpgrade(Stream stream, out SecurityMessagePr
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SecurityNegotiationException(
                     SR.Format(SR.NegotiationFailedIO, ioException.Message), ioException));
             }
-            catch (Exception exception)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SecurityNegotiationException(
```

Comment:
I would prefer something similar to:
```csharp
if(exception.GetType() != typeof(Exception))
{
    rethrow;
}
throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SecurityNegotiationException(
    exception.Message, exception));
```
In .Net Native, the WinRT api's really do just throw a generic Exception in which case we want to wrap it. Otherwise let the specific exception type bubble up.